src/wayland/shell/xdg/mod.rs

```diff
@@ -1115,6 +1123,30 @@ impl ToplevelSurface {
             .unwrap(),
         )
     }
+
+    /// Returns true if the provided surface is the parent of this toplevel surface.
+    pub fn is_parent(&self, parent: &WlSurface) -> bool {
+        match &self.shell_surface {
+            ToplevelKind::Xdg(toplevel) => xdg_handlers::is_parent(toplevel, parent),
+            ToplevelKind::ZxdgV6(toplevel) => zxdgv6_handlers::is_parent(toplevel, parent),
+        }
+    }
```

Comment:
It'd be better to have a method that returns the parent surface directly, that would then be checked for equality.